vassal-app/src/main/java/VASSAL/counters/Deck.java

```diff
@@ -1047,9 +1051,12 @@ public void setState(String state) {
 
   public Command setContentsFaceDown(boolean value) {
     final ChangeTracker t = new ChangeTracker(this);
-    final Command c = new NullCommand();
+    final   Command c = new NullCommand();
     faceDown = value;
-    return t.getChangeCommand().append(c).append(reportCommand(faceDownMsgFormat, value ? Resources.getString("Deck.face_down") : Resources.getString("Deck.face_up"))); //$NON-NLS-1$ //$NON-NLS-2$
+    if (Map.isChangeReportingEnabled()) {
+      c.append(reportCommand(faceDownMsgFormat, value ? Resources.getString("Deck.face_down") : Resources.getString("Deck.face_up")));
```

Comment:
Try and follow the pattern

c = c.append(newCommand);

as this will cause an appened non-null command to replace an existing NullCommand. Removes spurious NullCommands ending up in the command stream.

It's not needed if you know c will contain a non-null command (e.g. shuffle where it is initialised with setContents(l), but should always be used when you have

Command c = new NullCommand();
if (blah) {
  c = c.append(...);
}

Reply:
Okay I think I've got it right now, if I'm understanding you correctly.